packages/@ember/-internals/glimmer/lib/renderer.ts

```diff
@@ -398,6 +399,7 @@ export abstract class Renderer {
     let { _roots: roots, _env: env, _removedRoots: removedRoots } = this;
     let globalShouldReflush = false;
     let initialRootsLength: number;
+    let root: RootState, shouldReflush: boolean;
```

Comment:
I pulled them out specifically because I was hitting the hoisting assertion (would cause a new closure). I can try to figure out a way to put them back inline, maybe by moving the `runInAutotrackingTransaction` function up above like you suggest below.

Reply:
I guess this is fine, but the fact that they are hoisted and not reset is troublesome. Can you set them both to `null` or `undefined` right inside the loop?

Reply:
It's a lot of code to do that, figured it's better to just use `bind` for this.